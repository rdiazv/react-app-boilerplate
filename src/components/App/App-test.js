import {Â React, mount } from 'helpers/test-helper'
import App from 'components/App'

describe('App', () => {
  it('should render', () => {
    const component = mount(<App />)

    expect(component).toMatchSnapshot()
  })
})
