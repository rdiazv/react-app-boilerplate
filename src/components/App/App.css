@import 'theme/variables';

.App {
  color: $primary-color;
}
