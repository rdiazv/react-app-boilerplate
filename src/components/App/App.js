import React from 'react'

const App = () =>
  <div>react-app-boilerplate</div>

export default App
