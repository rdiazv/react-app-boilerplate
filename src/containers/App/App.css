@import 'theme/variables';

.App {
  padding: 32px;
}

.App__title {
  color: $primary-color;
  margin: 0;
}

.App__navigation {
  margin: 16px 0;
  display: flex;

  a {
    display: block;
    padding: 8px 16px;
    background-color: #eee;
    color: $text-color;
    text-decoration: none;
  }

  a:hover {
    background-color: #ddd;
  }

  a.active {
    background-color: $primary-color;
    color: #fff;
  }

  a + a {
    margin-left: 8px;
  }
}
