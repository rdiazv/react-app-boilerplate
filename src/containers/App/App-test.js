import {Â React, mount } from 'helpers/test-helper'
import App from 'containers/App'

describe('App', () => {
  it('should render', () => {
    const component = mount(<App />)

    expect(component).toMatchSnapshot()
  })
})
