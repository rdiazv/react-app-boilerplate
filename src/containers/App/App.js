import React from 'react'
import { BrowserRouter, Route, NavLink, Switch } from 'react-router-dom'
import Home from 'containers/Home'
import About from 'containers/About'
import './App.css'

const App = () =>
  <BrowserRouter>
    <div className="App">
      <h1 className="App__title">react-app-boilerplate</h1>

      <div className="App__navigation">
        <NavLink exact to="/">Home</NavLink>
        <NavLink to="/about">About</NavLink>
      </div>

      <div className="App__contents">
        <Switch>
          <Route
            component={Home}
            exact
            path="/"
          />
          <Route
            component={About}
            path="/about"
          />
        </Switch>
      </div>
    </div>
  </BrowserRouter>

export default App
