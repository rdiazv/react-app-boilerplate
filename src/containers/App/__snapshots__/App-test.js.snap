// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": undefined,
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="App"
      >
        <h1
          className="App__title"
        >
          react-app-boilerplate
        </h1>
        <div
          className="App__navigation"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/"
          >
            <Route
              exact={true}
              path="/"
            >
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            activeClassName="active"
            to="/about"
          >
            <Route
              path="/about"
            >
              <Link
                replace={false}
                to="/about"
              >
                <a
                  href="/about"
                  onClick={[Function]}
                >
                  About
                </a>
              </Link>
            </Route>
          </NavLink>
        </div>
        <div
          className="App__contents"
        >
          <Switch />
        </div>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;
